File: Atcoder_#332/javascript/b_glass_ans_mug.js
function main(input) {
  let [k, g, m] = input.trim().split(' ').map(Number);

  let glass = 0;
  let mug = 0;

  for (let i = 0; i < k; i++) {
      if (glass === g) {
          glass = 0;
      } else if (mug === 0) {
          mug = m;
      } else {
          let transfer = Math.min(mug, g - glass);
          mug -= transfer;
          glass += transfer;
      }
  }

  console.log(glass, mug);
}

main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: Atcoder_#332/ruby/b_glass_ans_mug.rb
k, g, m = gets.split(" ").map(&:to_i)

glass = 0
mug = 0

k.times do |i|
  if glass == g
    glass = 0
  elsif mug == 0
    mug = m
  else
    transfer = [mug, g - glass].min
    mug -= transfer
    glass += transfer
  end
end

puts "#{glass} #{mug}"
